earth_enterprise/src/fusion/autoingest/sysman/StateUpdater.cpp

```diff
@@ -276,7 +279,9 @@ void StateUpdater::SetStateForRefAndDependents(
     function<bool(AssetDefs::State)> updateStatePredicate) {
   try {
     SharedString verref = AssetVersionImpl::Key(ref);
```

Comment:
I have to admit being a little confused about what exactly is going on here. As far as I can tell, `ref` can come in several formats: an asset, a version, or it can ask for the "current" or "lastgood" version of a specific asset. The `Key` function sort of normalizes it so that "verref" always refers to a specific version number. I think that ref and verref are the same in some cases, but not all.

As far as whether `Bind` loads the version into the cache, I'm not certain about that. The code gets hard to follow pretty quickly. In some cases it has to load the asset (not the version) into the cache so that it can figure out which version is current or lastgood. There may be a naming issue here, but ultimately it does what it says - it returns the normalized key that you can use to refer to a specific asset version.

Reply:
Ok, I didn't catch the normalization aspect of `Key`.